lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py

```diff
@@ -0,0 +1,459 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Alertmanager Remote Configuration library.
+
+This library offers the option of configuring Alertmanager via relation data.
+It has been created with the `alertmanager-k8s` and the `alertmanager-k8s-configurer`
+(https://charmhub.io/alertmanager-configurer-k8s) charms in mind, but can be used by any charms
+which require functionalities implemented by this library.
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.alertmanager_k8s.v0.alertmanager_remote_configuration
+```
+
+Charms that need to push Alertmanager configuration to a charm exposing relation using
+the `alertmanager_remote_configuration` interface, should use the `RemoteConfigurationProvider`.
+Charms that need to can utilize the Alertmanager configuration provided from the external source
+through a relation using the `alertmanager_remote_configuration` interface, should use
+the `RemoteConfigurationRequirer`.
+"""
+
+import json
+import logging
+from pathlib import Path
+from typing import Optional, Tuple
+
+import yaml
+from ops.charm import CharmBase, RelationJoinedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import Relation
+
+# The unique Charmhub library identifier, never change it
+LIBID = "something dummy for now"
```

Comment:
Now that we're set on `alertmanager_remote_configuration` I registered it:
```shell
$ grep "LIBID" lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py
LIBID = "0e5a4c0ecde34c9880bb8899ac53444d"
```

Reply:
Thanks!
Updated in the PR.